build-plugin/action.yml

```
@@ -0,0 +1,148 @@
+        echo "plugin-type=${GRAFANA_PLUGIN_TYPE}" >> $GITHUB_OUTPUT
+        echo "archive=${GRAFANA_PLUGIN_ARTIFACT}" >> $GITHUB_OUTPUT
+        echo "archive-sha1sum=${GRAFANA_PLUGIN_ARTIFACT_SHA1SUM}" >> $GITHUB_OUTPUT
+
+        echo "github-tag=${GITHUB_REF#refs/*/}" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: Read changelog
+      id: changelog
+      run: |
+        awk '/^## / {s++} s == 1 {print}' CHANGELOG.md > release_notes.md
+        echo "path=release_notes.md" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: Check package version
+      run: if [ "v${{ steps.metadata.outputs.plugin-version }}" != "${{ steps.metadata.outputs.github-tag }}" ]; then printf "\033[0;31mPlugin version doesn't match tag name\033[0m\n"; exit 1; fi
+      shell: bash
+
+    - name: Package plugin
+      id: package-plugin
+      run: |
+        mv dist ${{ steps.metadata.outputs.plugin-id }}
+        zip ${{ steps.metadata.outputs.archive }} ${{ steps.metadata.outputs.plugin-id }} -r
+        sha1sum ${{ steps.metadata.outputs.archive }} | cut -f1 -d' ' > ${{ steps.metadata.outputs.archive-sha1sum }}
+      shell: bash
+
+    - name: Validate plugin
+      run: |
+        git clone https://github.com/grafana/plugin-validator
+        pushd ./plugin-validator/pkg/cmd/plugincheck2
+        go install
+        popd
+        plugincheck2 -config ./plugin-validator/config/default.yaml ${{ steps.metadata.outputs.archive }}
+      shell: bash
+
+    - name: Create Github release
+      uses: softprops/action-gh-release@v1
+      with:
+        draft: true
+        generate_release_notes: true
+        token: ${{ inputs.token }}
+        files: |
+          ./${{ steps.metadata.outputs.archive }}
+          ./${{ steps.metadata.outputs.archive-sha1sum }}
+        body: |
+          **This GitHub draft release has been created for your plugin.**
+
+          _Note: if this is the first release for your plugin, consult the [distributing-your-plugin section](https://github.com/${{github.repository}}/blob/main/README.md#distributing-your-plugin) of the README.
+
+          If you want to submit this release to Grafana for publication in our catalog, consider the following steps:
+
+          - Check the Validate plugin step in the [release workflow](https://github.com/${{github.repository}}/commit/${{github.sha}}/checks/${{github.run_id}}) for any warnings that need attention
+          - Go to https://grafana.com/auth/sign-in/ to sign into your account.
+          - Once logged in, click **My Plugins** on the navigation menu.
+          - Click **Submit Plugin**.
+          - Fill in the Plugin Submission form:
+            - Paste this [.zip asset link](https://github.com/${{ github.repository }}/releases/download/v${{ steps.metadata.outputs.plugin-version }}/${{ steps.metadata.outputs.archive }}) in the Plugin URL field
+            - Paste this [.zip.sha1 link](https://github.com/$/releases/download/v$/$) in the SHA1 field
```

What are the `$` signs interpreted as in this link? Am I missing something or should it be the following?
```suggestion
            - Paste this [.zip.sha1 link](https://github.com/${{ github.repository }}/releases/download/v${{ steps.metadata.outputs.plugin-version }}/${{ steps.metadata.outputs.archive-sha1sum }}) in the SHA1 field
```

---

I copy pasted this exactly as we distribute it in our release template so I can't really answer how they work. I think your suggestion makes much more sense.

---

🤷‍♂️